vengine/vengine/resource_management/ResourceManager.cpp

```diff
@@ -100,6 +100,81 @@ uint32_t ResourceManager::addMesh(
     return meshes.size() - 1;
 }
 
+uint32_t ResourceManager::addMesh(std::string meshPath, MeshData meshData)
+{
+	// No mesh imported, send default mesh back
+	if (meshData.vertexStreams.positions.size() == 0)
+	{
+		return 0;
+	}
+
+	// Smooth normals in mesh data
+	// MeshDataModifier::smoothNormals(meshData);
+
+	auto map_iterator = this->meshPaths.find(meshPath);
+	if (this->meshPaths.end() == map_iterator)
+	{
+        //No mesh exists, create new
+		this->meshPaths.insert({meshPath, this->meshes.size()});
+
+		// NOTE: meshes.size as key only works if we never remove resources the map...
+		// Create mesh, insert into map of meshes
+		meshes.insert({meshes.size(), meshLoader.createMesh(meshData)});
+	}
+	else
+	{
+		uint32_t meshID = map_iterator->second;
+        auto mesh_iterator = this->meshes.find(meshID);
+        dev->waitIdle();
```

Comment:
Can you create a "safeCleanup()" function in the Mesh class, containing the device.waitIdle() and regular cleanup(), and then use this safeCleanup() inside the ResourceManager?